modules/apps/friendly-url/friendly-url-api/src/main/java/com/liferay/friendly/url/separator/util/FriendlyURLSeparatorUtil.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.friendly.url.separator.util;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.exception.LayoutFriendlyURLException;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.portlet.FriendlyURLResolver;
+import com.liferay.portal.kernel.portlet.FriendlyURLResolverRegistryUtil;
+import com.liferay.portal.kernel.service.LayoutFriendlyURLLocalServiceUtil;
+import com.liferay.portal.kernel.util.FriendlyURLNormalizer;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.service.impl.LayoutLocalServiceHelper;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * @author Carolina Barbosa
+ */
+public class FriendlyURLSeparatorUtil {
+
+	public static String normalize(
+		FriendlyURLNormalizer friendlyURLNormalizer,
+		String friendlyURLSeparator) {
+
+		return StringUtil.quote(
+			friendlyURLNormalizer.normalizeWithPeriodsAndSlashes(
+				friendlyURLSeparator),
+			StringPool.SLASH);
+	}
+
+	public static String validate(
+		long companyId, FriendlyURLNormalizer friendlyURLNormalizer,
+		String friendlyURLResolverKey, String friendlyURLSeparator,
+		List<String> friendlyURLSeparators,
+		LayoutLocalServiceHelper layoutLocalServiceHelper, Locale locale) {
+
+		if (Validator.isNull(friendlyURLSeparator)) {
+			return LanguageUtil.get(
+				locale, "friendly-url-separator-error-cannot-be-empty");
+		}
+
+		friendlyURLSeparator = normalize(
+			friendlyURLNormalizer, friendlyURLSeparator);
```

Comment:
Is there a way to only normalize the separator once? With these changes it is normalized here and then normalized again in the calling method prior to saving.

Reply:
@jonathanmccann I updated the code and now it is normalizing only once